contracts/Everlong.sol

```diff
@@ -64,113 +62,16 @@ import { IHyperdrive } from "hyperdrive/contracts/src/interfaces/IHyperdrive.sol
 /// @custom:disclaimer The language used in this code is for coding convenience
 ///                    only, and is not intended to, and does not, have any
 ///                    particular legal or regulatory significance.
-contract Everlong is Admin, ERC4626, Positions {
-    /// @notice Virtual shares are used to mitigate inflation attacks.
-    bool public constant useVirtualShares = true;
-
-    /// @notice Used to reduce the feasibility of an inflation attack.
-    /// TODO: Determine the appropriate value for our case. Current value
-    ///       was picked arbitrarily.
-    uint8 public constant decimalsOffset = 3;
-
-    /// @notice Address of the Hyperdrive instance wrapped by Everlong.
-    address public immutable hyperdrive;
-
-    /// @notice ERC20 token used for deposits, idle liquidity, and
-    ///         the purchase of bonds from the Hyperdrive instance.
-    address internal immutable _asset;
-
-    /// @notice Decimals used by the `_asset`.
-    uint8 internal immutable _decimals;
-
-    /// @notice Name of the Everlong token.
-    string internal _name;
-
-    /// @notice Symbol of the Everlong token.
-    string internal _symbol;
-
+contract Everlong is EverlongAdmin, EverlongPositions {
     /// @notice Initial configuration paramters for Everlong.
-    /// @param hyperdrive_ Address of the Hyperdrive instance wrapped by Everlong.
     /// @param name_ Name of the ERC20 token managed by Everlong.
     /// @param symbol_ Symbol of the ERC20 token managed by Everlong.
-    /// @param asset_ ERC20 token used to purchase bonds from the Hyperdrive instance.
+    /// @param hyperdrive_ Address of the Hyperdrive instance wrapped by Everlong.
+    /// @param asBase_ Whether to use Hyperdrive's base token for bond purchases.
     constructor(
         string memory name_,
         string memory symbol_,
         address hyperdrive_,
```

Comment:
Looks like I missed this. Can we rename these params too? I will add a note to the style guide once I finish this review.

Reply:
Ahh shoot me too, will do